client/src/store/stores/route.ts

```diff
@@ -42,7 +51,43 @@ export default class RouteStore {
     return pp;
   }
 
-  @action.bound private dropCache() {
+  @computed get namespace(): string | null {
+    return this.pathParts[0] || null;
+  }
+
+  @action.bound
+  navigate(to: string, options?: NavigateOptions<{}>) {
+    return this.history.navigate(to, options);
+  }
+
+  @action.bound
+  private dropCache() {
     this.cache = {};
   }
+
+  private listen() {
+    this.history.listen(({ location, action }) => {
+      // console.log('in history listen: ', location, action);
+      runInAction(() => {
+        this.dropCache();
+        this.location = location;
+      });
+    });
+
+    autorun(() => {
+      if (this.namespace) {
+        localStorage.setItem(RouteStore.NAMESPACE_LS_KEY, this.namespace);
+      }
+    });
+  }
+
+  private restoreNamespace() {
+    const currentNamespace = this.location.pathname.split('/').slice(1)[0];
```

Comment:
Why not to use [this.namespace](https://github.com/cilium/hubble-ui/pull/41/files#diff-4ce5359db07a4930a345b702ab4adb8bR54) ?